File: src/main/java/com/refactortalentscrm/service/CandidateService.java
package com.refactortalentscrm.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.refactortalentscrm.domain.Candidate;
import com.refactortalentscrm.dto.CandidateDTO;
import com.refactortalentscrm.dto.CandidateSearchDTO;
import com.refactortalentscrm.dto.CandidatePageDTO;
import com.refactortalentscrm.dto.CandidateConvertCriteriaDTO;
import com.refactortalentscrm.service.GenericService;
import com.refactortalentscrm.dto.common.RequestDTO;
import com.refactortalentscrm.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CandidateService extends GenericService<Candidate, Integer> {

	List<Candidate> findAll();

	ResultDTO addCandidate(CandidateDTO candidateDTO, RequestDTO requestDTO);

	ResultDTO updateCandidate(CandidateDTO candidateDTO, RequestDTO requestDTO);

    Page<Candidate> getAllCandidates(Pageable pageable);

    Page<Candidate> getAllCandidates(Specification<Candidate> spec, Pageable pageable);

	ResponseEntity<CandidatePageDTO> getCandidates(CandidateSearchDTO candidateSearchDTO);
	
	List<CandidateDTO> convertCandidatesToCandidateDTOs(List<Candidate> candidates, CandidateConvertCriteriaDTO convertCriteria);

	CandidateDTO getCandidateDTOById(Integer candidateId);







}

File: refactortalentscrm_ui/src/services/CandidateService.js
import http from "../http-common";

class CandidateService {
  getAllCandidates(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/candidate/candidates`, searchDTO);
  }

  get(candidateId) {
    return this.getRequest(`/candidate/${candidateId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/candidate?field=${matchData}`, null);
  }

  addCandidate(data) {
    return http.post("/candidate/addCandidate", data);
  }

  update(data) {
  	return http.post("/candidate/updateCandidate", data);
  }
  
  uploadImage(data,candidateId) {
  	return http.postForm("/candidate/uploadImage/"+candidateId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CandidateService();
